helion/_compiler/inductor_lowering.py

```diff
@@ -172,29 +185,49 @@ def convert_arg(arg: Node) -> TensorBox:
                 new_node.kwargs = {**new_node.kwargs, "_extra_args": [*nodes]}
         else:
             new_node = create_extra_node(node, buffer, [*node._input_nodes, *nodes])
+
+        # Store output index if this buffer corresponds to an output
+        if buffer.get_name() in buffer_name_to_output_index:
+            new_node.meta["output_index"] = buffer_name_to_output_index[
+                buffer.get_name()
+            ]
+
         lowering_cls = (
             PointwiseLowering
             if isinstance(buffer.data, Pointwise)
             else ReductionLowering
         )
         buffer.freeze_layout()
+
+        current_input_nodes = new_node._input_nodes
+        current_input_names = []
+        for inp_node in current_input_nodes:
+            current_input_names.append(node_to_buf_name_mapping[inp_node])
+
         used_input_names = strip_unused_inputs(
             new_node,
             buffer.get_read_names(),
-            dict(
-                zip(
-                    node.all_input_nodes,
-                    [*input_names, *extra_input_names],
-                    strict=True,
-                )
-            ),
```

Comment:
I updated my initial implementation to make it more clean. The problem that necessitated this change is that for `var_mean` op, with original code we got these two which don't have the same length (thus `strict=True` fails):
```
node.all_input_nodes: [_phi]
[*input_names, *extra_input_names]: ['var_mean_input0', 'buf0']
```
`_phi` node maps to name `var_mean_input0` as expected, but we don't have a node that maps to `buf0` (which is an extra buffer created by Inductor that we need to find the Helion-created extra FX node for).

With the updated way of building the mapping, we now have:
```
current_input_nodes: {_phi: None, var_mean_extra: None}
current_input_names: ['var_mean_input0', 'buf0']
```
And each Inductor buffer now have their corresponding FX node correctly tracked.